src/mpl/include/mpl_base.h

```diff
@@ -155,4 +155,15 @@ typedef bool _Bool;
 #define MPL_ROUND_UP_ALIGN(a, alignment) (((a) + ((alignment) - 1)) & (~((alignment) - 1)))
 #define MPL_ROUND_DOWN_ALIGN(a, alignment) ((a) & (~((alignment) - 1)))
 
+#define MPL_CAST(T, val) CAST_##T((val))
+#ifdef NDEBUG
+#define MPL_CAST_int(val) ((int) (val))
+#define MPL_CAST_Aint(val) ((MPI_Aint) (val))
+#else
+#define MPL_CAST_int(val) \
+    (((val) > INT_MAX || ((val) < 0 && (val) < INT_MIN)) ? (assert(0), 0) : (int) (val))
+#define MPL_CAST_Aint(val) \
+    (((val) > LONG_MAX || ((val) < 0 && (val) < LONG_MIN)) ? (assert(0), 0) : (MPI_Aint) (val))
```

Comment:
`MPI_Aint` may not be a `long`. We define `MPIR_AINT_MAX`, maybe just define these macros in `MPIR`- layer?

Reply:
Fixed  this. Added `MPIR_AINT_MIN` in `configure.ac` as well.